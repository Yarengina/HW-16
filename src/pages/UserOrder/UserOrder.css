.InputData {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 12px;
    margin-left: 20px;
}

.UserInputs {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    border: initial;
    padding: 0px;
    margin: 0px;
}

.UserInput {
    width: 95%;
    font-size: 16px;
    color: rgb(96 121 139);
    border: 1px solid rgb(153 153 153 / 61%);
    border-radius: 4px;
    padding: 4px 8px;
}

.ButtonBack {
    width: 100px;
    height: 26px;
    background-color: #ffba00;
    border: initial;
    border-radius: 4px;
    margin-top: 10px;
}

.ButtonSend {
    width: 60px;
    height: 26px;
    background-color: #4caf50;
    border: initial;
    border-radius: 4px;
    color: white;
    font-weight: bolder;
    margin-left: 6px;
}
