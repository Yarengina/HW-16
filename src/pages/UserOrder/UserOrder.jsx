// eslint-disable-next-line no-unused-vars
import * as React from 'react'
/* eslint-disable jsx-a11y/label-has-associated-control */
import { useRef, useContext, useState } from 'react'
import { useNavigate } from 'react-router-dom'
import UserInfoContext from '../../contexts/UserInfoContext'
import './UserOrder.css'

function UserOrder() {
    const { user, setUser } = useContext(UserInfoContext)

    const [userPhone, setUserPhone] = useState('')

    const onChangePhone = ({ target: { value } }) => {
        const valueInput = value.replace(/\D+/g, '')

        if (userPhone !== valueInput) {
            setUserPhone(valueInput)
        }
    }

    const emailInputRef = useRef(null)
    const phoneInputRef = useRef(null)
    const nameInputRef = useRef(null)

    const navigate = useNavigate()

    const goBack = () => {
        navigate('/order')
    }

    const handleOrder = () => {
        setUser({
            ...user,
            phone: phoneInputRef.current.value,
            name: nameInputRef.current.value,
        })
        navigate('/order/receipt')
    }

    return (
        <section className="InputData">
            <h1>Input data</h1>
            <div className="UserInputs">
                <label htmlFor="usermail">Email: </label>
                <input
                    data-testid="usermail"
                    className="UserInput"
                    id="usermail"
                    type="email"
                    defaultValue={user.email}
                    ref={emailInputRef}
                />
            </div>
            <div className="UserInputs">
                <label htmlFor="userphone">Phone: </label>
                <input
                    data-testid="userphone"
                    className="UserInput"
                    id="userphone"
                    type="tel"
                    onChange={onChangePhone}
                    value={userPhone}
                    ref={phoneInputRef}
                />
            </div>
            <div className="UserInputs">
                <label htmlFor="username">Name: </label>
                <input
                    className="UserInput"
                    id="username"
                    type="text"
                    ref={nameInputRef}
                />
            </div>
            <div>
                <button className="ButtonBack" type="button" onClick={goBack}>
                    Back to cart
                </button>
                <button
                    className="ButtonSend"
                    type="button"
                    onClick={handleOrder}
                >
                    Send
                </button>
            </div>
        </section>
    )
}

export default UserOrder
