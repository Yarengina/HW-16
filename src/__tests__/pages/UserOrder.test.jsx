// eslint-disable-next-line no-unused-vars
import * as React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { render } from '../utils/renderWithRouter'
import UserOrder from '../../pages/UserOrder/UserOrder'

describe('<UserOrder />', () => {
    it('should render successfully and match snapshot', () => {
        const { container } = render(<UserOrder />)
        expect(container).toMatchSnapshot()
    })

    it('should be able to type a phone', () => {
        render(<UserOrder />)
        const inputPhone = screen.getByRole('textbox', { name: /phone:/i })
        expect(inputPhone.value).toBe('')

        userEvent.type(inputPhone, '123')
        expect(inputPhone.value).toBe('123')
    })

    it('should be able to type only numbers', () => {
        render(<UserOrder />)
        const inputPhone = screen.getByRole('textbox', { name: /phone:/i })
        userEvent.type(inputPhone, 'user')
        expect(inputPhone.value).toBe('')
    })
})
