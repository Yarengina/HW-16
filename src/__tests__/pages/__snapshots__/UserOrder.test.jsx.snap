// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserOrder /> should render successfully and match snapshot 1`] = `
<div>
  <section
    class="InputData"
  >
    <h1>
      Input data
    </h1>
    <div
      class="UserInputs"
    >
      <label
        for="usermail"
      >
        Email: 
      </label>
      <input
        class="UserInput"
        data-testid="usermail"
        id="usermail"
        type="email"
        value=""
      />
    </div>
    <div
      class="UserInputs"
    >
      <label
        for="userphone"
      >
        Phone: 
      </label>
      <input
        class="UserInput"
        data-testid="userphone"
        id="userphone"
        type="tel"
        value=""
      />
    </div>
    <div
      class="UserInputs"
    >
      <label
        for="username"
      >
        Name: 
      </label>
      <input
        class="UserInput"
        id="username"
        type="text"
      />
    </div>
    <div>
      <button
        class="ButtonBack"
        type="button"
      >
        Back to cart
      </button>
      <button
        class="ButtonSend"
        type="button"
      >
        Send
      </button>
    </div>
  </section>
</div>
`;
